lib/index.d.ts

```diff
@@ -39,14 +40,40 @@ declare namespace Stardog {
         password: string;
     }
 
+    // Kind of a hack, but necessary to get around the way TS libs define the `Request` object.
+    type RequestConstructor = {
+      new (input: string | Request, init?: RequestInit): Request;
+    };
+
+    type RequestCreator<Constructor = RequestConstructor, ReturnType = string | Request> = ({ uri, Request }: { uri: string; Request: Constructor }) => ReturnType;
+
+    /** Optional meta-configuration for a Connection */
+    interface ConnectionMeta {
+        /**
+         * If defined, this method is called whenever a `fetch` call is about
+         * to be made for a given Connection. The method should return a
+         * `Request` object (either a browser `Request` or a `node-fetch`
+         * `Request`) or a string URI, which will be passed as the first
+         * argument to the imminent `fetch` call. This is useful in cases where
+         * fetch needs to allow connections with self-signed certificates, for
+         * example.
+         *
```

Comment:
I'm sure this is very descriptive, but I still found it a little confusing. Could be my lack of JS-fu. It also might not matter since it's all optional

Reply:
I tried updating this a bit to make it clearer.